### PR TITLE
GEODE-874: Place all build-from-source instructions into file BUILDIN…

### PR DESCRIPTION
**Pull request by karensmolermiller**
GEODE-874: Place all build-from-source instructions into file BUILDING.md, update the README.md file to tell people to look in the BUILDING.md file if appropriate, update the Geode-In-5-Minutes to not include building instructions, and remove irrelevant files COMPILING.txt and RUNNING.txt from the repo. Reword the README.md claim that Geode can do rolling updates between major versions, as there are no major versions, and we can't make promises about the future.

### CODE REVIEW

**Comment by metatype** [comment]
+1

If you follow the commit comment conventions at [1] our commit history will be more readable and consistent.  Thanks!

[1] http://chris.beams.io/posts/git-commit/

**Comment by pdxrunner** [comment]
+1

A note on BUILDING.md - Should the verification steps reinforce that the GEODE install bin directory (geode-assembly\build\install\apache-geode\bin) be placed in the PATH. Otherwise, the steps as shown will fail.

**Comment by karensmolermiller** [comment]
I do not believe that those building Geode from source should be reminded to place a bin directory
in their path.  I think that it is enough, in this instance, to identify the 2 locations of the built
gfsh script and let the developer decide how to invoke the script.

File: README.md [hunk 1]
```diff
@@ -36,19 +37,18 @@ Apache Geode includes the following features:
 * Cluster-wide configurations that can be persisted and exported to other clusters.
 * Remote cluster management through HTTP.
 * REST APIs for REST-enabled application development.
-* Rolling upgrade between major version releases.
+* Rolling upgrades may be possible, but they will be subject to any limitations imposed by new features.
```
Comment (sbawaska) [inline README.md]: We have special handling of messages sent from members of older versions. I think it is safe to leave this as is.

**Comment by karensmolermiller** [comment]
I've revised the BUILDING.md file such that the redundant installDist command line option is no longer included.

I have not changed the Windows build instructions to suggest using the gradlew.bat script.  The instructions modified in this PR are those that should be followed by someone who downloads source from the Releases page of the http://geode.incubator.apache.org/ site. The source from that location does not include a Windows gradlew.bat script, per Apache rules.  I suggest that instructions for building from source when acquiring source from GitHub belong on the cwiki; the gradlew.bat script is still included in the apache/incubator-geode repository.  For the longer term, I further suggest that the Windows gradlew.bat script be rewritten such that it can be included in the source of a release.  Then, there can be a single set of build instructions, and the build instructions become independent of the location of the source download.

**Comment by upthewaterspout** [comment]
I'll merge this.

**Pull request closed without merging**